EapWebInterface/src/main/java/de/hpi/unicorn/application/pages/input/generator/EventGenerator.java

```diff
@@ -121,278 +111,53 @@ public void generateEvents(int eventCount, int scaleFactor, EapEventType eventTy
         if (!attributeDependencyManager.isBaseAttributeInAnyDependency(baseAttribute)) {
             return;
         }
 
-        String baseAttributeInput;
-        if (baseAttribute.getType() == AttributeTypeEnum.DATE) {
-            baseAttributeInput = dateFormatter.format(eventValues.get(baseAttribute.getName()));
-        }
-        else {
-            baseAttributeInput = String.valueOf(eventValues.get(baseAttribute.getName()));
-        }
-
+        // iterate over each dependency defined for the base-attribute
         for (AttributeDependency attributeDependency : attributeDependencyManager.getAttributeDependencies(baseAttribute)) {
             TypeTreeNode dependentAttribute = attributeDependency.getDependentAttribute();
-            List<String> possibleDependentValues = new ArrayList<>();
-            for (AttributeValueDependency attributeValueDependency
-                    : attributeDependencyManager.getAttributeValueDependencies(attributeDependency)) {
-                if (isInRange(baseAttributeInput, attributeValueDependency.getBaseAttributeValue(), baseAttribute.getType())) {
-                    possibleDependentValues.add(attributeValueDependency.getDependentAttributeValues());
+            List<AttributeInput> possibleDependentAttributeInputs = new ArrayList<>();
+            // iterate over each value-dependency defined on this dependency
+            for (AttributeValueDependency attributeValueDependency : attributeDependencyManager.getAttributeValueDependencies
+                    (attributeDependency)) {
+                // if the (already computed) value for the base attribute is within the value range of the current value-dependency-rule for the
+                // base attribute (hence this dependency-rule can be applied), add the values of the corresponding dependent-attribute to the list of
+                // possible values.
+                if (baseAttributeInput.isInRange(attributeValueDependency.getBaseAttributeValue())) {
+                    AttributeInput possibleDependentAttributeInput = AttributeInput.attributeInputFactory(dependentAttribute);
+                    possibleDependentAttributeInput.setInput(attributeValueDependency.getDependentAttributeValues());
+                    possibleDependentAttributeInputs.add(possibleDependentAttributeInput);
                 }
             }
-            if (!possibleDependentValues.isEmpty()) {
-                String possibleDependentValue = possibleDependentValues.get(getRandomIndex(possibleDependentValues));
-                setRandomValueFromRangeForAttribute(dependentAttribute, possibleDependentValue, eventValues);
-                tryToFillDependentAttributes(dependentAttribute, eventValues);
+            if (!possibleDependentAttributeInputs.isEmpty()) {
+                AttributeInput dependentAttributeInput = possibleDependentAttributeInputs.get(AttributeInput.getRandomIndex
+                        (possibleDependentAttributeInputs));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') '(' should be on the previous line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=checkstyle%3AMethod_Param_Pad_BP2016)